lib/hooks.dart

```diff
@@ -102,6 +102,60 @@ StateHook<T> useState<T>(T initialValue) => StateHook(initialValue);
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#lazy-initial-state>.
 StateHook<T> useStateLazy<T>(T init()) => StateHook.lazy(init);
 
+/// Runs [sideEffect] after every completed render of a [DartFunctionComponent].
+///
+/// If [dependencies] are given, [sideEffect] will only run if one of the [dependencies] have changed.
+/// [sideEffect] may return a cleanup function that is run before the component unmounts or re-renders.
+///
+/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
+/// >
+/// > * Only call Hooks at the top level.
+/// > * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// __Example__:
+///
+/// ```
+/// UseEffectTestComponent(Map props) {
+///   final count = useState(1);
+///   final evenOdd = useState('even');
+///
+///   useEffect(() {
+///     if (count.value % 2 == 0) {
+///       evenOdd.set('even');
+///     } else {
+///       evenOdd.set('odd');
+///     }
+///     return () {
+///       print('count is changing...');
```

Comment:
Is this the "cleanup" function? Might be good to notate that either with a comment, or update the string here to say something like: 

```suggestion
///       print('count is changing... do some cleanup if you need to');
```